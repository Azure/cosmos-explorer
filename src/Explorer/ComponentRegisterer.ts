import * as ko from "knockout";
import { DiffEditorComponent } from "./Controls/DiffEditor/DiffEditorComponent";
import { DynamicListComponent } from "./Controls/DynamicList/DynamicListComponent";
import { EditorComponent } from "./Controls/Editor/EditorComponent";
import { JsonEditorComponent } from "./Controls/JsonEditor/JsonEditorComponent";
import { ThroughputInputComponentAutoPilotV3 } from "./Controls/ThroughputInput/ThroughputInputComponentAutoPilotV3";
import * as PaneComponents from "./Panes/PaneComponents";

ko.components.register("editor", new EditorComponent());
ko.components.register("json-editor", new JsonEditorComponent());
ko.components.register("diff-editor", new DiffEditorComponent());
ko.components.register("dynamic-list", DynamicListComponent);
ko.components.register("throughput-input-autopilot-v3", ThroughputInputComponentAutoPilotV3);

// Panes
ko.components.register("cassandra-add-collection-pane", new PaneComponents.CassandraAddCollectionPaneComponent());
