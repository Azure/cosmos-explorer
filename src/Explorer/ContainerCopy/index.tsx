import { MonitorCopyJobsRefState } from "Explorer/ContainerCopy/MonitorCopyJobs/MonitorCopyJobRefState";
import React, { useEffect } from "react";
import CopyJobCommandBar from "./CommandBar/CopyJobCommandBar";
import "./containerCopyStyles.less";
import MonitorCopyJobs, { MonitorCopyJobsRef } from "./MonitorCopyJobs/MonitorCopyJobs";
import { ContainerCopyProps } from "./Types";

const ContainerCopyPanel: React.FC<ContainerCopyProps> = ({ container }) => {
  const monitorCopyJobsRef = React.useRef<MonitorCopyJobsRef>();
  useEffect(() => {
    if (monitorCopyJobsRef.current) {
      MonitorCopyJobsRefState.getState().setRef(monitorCopyJobsRef.current);
    }
  }, [monitorCopyJobsRef.current]);
  return (
    <div id="containerCopyWrapper" className="flexContainer hideOverflows">
      <CopyJobCommandBar container={container} />
      <MonitorCopyJobs ref={monitorCopyJobsRef} />
    </div>
  );
};

export default ContainerCopyPanel;
