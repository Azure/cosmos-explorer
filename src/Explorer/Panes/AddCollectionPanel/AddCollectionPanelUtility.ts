import { userContext } from "UserContext";

export function getPartitionKeyTooltipText(): string {
  if (userContext.apiType === "Mongo") {
    return "The shard key (field) is used to split your data across many replica sets (shards) to achieve unlimited scalability. Itâ€™s critical to choose a field that will evenly distribute your data.";
  }

  let tooltipText = `The ${getPartitionKeyName(
    true,
  )} is used to automatically distribute data across partitions for scalability. Choose a property in your JSON document that has a wide range of values and evenly distributes request volume.`;

  if (userContext.apiType === "SQL") {
    tooltipText += " For small read-heavy workloads or write-heavy workloads of any size, id is often a good choice.";
  }

  return tooltipText;
}

export function getPartitionKeyName(isLowerCase?: boolean): string {
  const partitionKeyName = userContext.apiType === "Mongo" ? "Shard key" : "Partition key";

  return isLowerCase ? partitionKeyName.toLocaleLowerCase() : partitionKeyName;
}

export function getPartitionKeyPlaceHolder(index?: number): string {
  switch (userContext.apiType) {
    case "Mongo":
      return "e.g., categoryId";
    case "Gremlin":
      return "e.g., /address";
    case "SQL":
      return `${
        index === undefined
          ? "Required - first partition key e.g., /TenantId"
          : index === 0
          ? "second partition key e.g., /UserId"
          : "third partition key e.g., /SessionId"
      }`;
    default:
      return "e.g., /address/zipCode";
  }
}

export function getPartitionKey(isQuickstart?: boolean): string {
  if (userContext.apiType !== "SQL" && userContext.apiType !== "Mongo") {
    return "";
  }
  if (userContext.features.partitionKeyDefault) {
    return userContext.apiType === "SQL" ? "/id" : "_id";
  }
  if (userContext.features.partitionKeyDefault2) {
    return userContext.apiType === "SQL" ? "/pk" : "pk";
  }
  if (isQuickstart) {
    return userContext.apiType === "SQL" ? "/categoryId" : "categoryId";
  }
  return "";
}
