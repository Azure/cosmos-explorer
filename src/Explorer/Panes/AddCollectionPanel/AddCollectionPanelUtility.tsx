import { DirectionalHint, Icon, Link, Stack, Text, TooltipHost } from "@fluentui/react";
import * as Constants from "Common/Constants";
import { configContext, Platform } from "ConfigContext";
import * as DataModels from "Contracts/DataModels";
import { getFullTextLanguageOptions } from "Explorer/Controls/FullTextSeach/FullTextPoliciesComponent";
import { isFabricNative } from "Platform/Fabric/FabricUtil";
import React from "react";
import { userContext } from "UserContext";

export function getPartitionKeyTooltipText(): string {
  if (userContext.apiType === "Mongo") {
    return "The shard key (field) is used to split your data across many replica sets (shards) to achieve unlimited scalability. Itâ€™s critical to choose a field that will evenly distribute your data.";
  }

  let tooltipText = `The ${getPartitionKeyName(
    true,
  )} is used to automatically distribute data across partitions for scalability. Choose a property in your JSON document that has a wide range of values and evenly distributes request volume.`;

  if (userContext.apiType === "SQL") {
    tooltipText += " For small read-heavy workloads or write-heavy workloads of any size, id is often a good choice.";
  }

  return tooltipText;
}

export function getPartitionKeyName(isLowerCase?: boolean): string {
  const partitionKeyName = userContext.apiType === "Mongo" ? "Shard key" : "Partition key";

  return isLowerCase ? partitionKeyName.toLocaleLowerCase() : partitionKeyName;
}

export function getPartitionKeyPlaceHolder(index?: number): string {
  switch (userContext.apiType) {
    case "Mongo":
      return "e.g., categoryId";
    case "Gremlin":
      return "e.g., /address";
    case "SQL":
      return `${
        index === undefined
          ? "Required - first partition key e.g., /TenantId"
          : index === 0
          ? "second partition key e.g., /UserId"
          : "third partition key e.g., /SessionId"
      }`;
    default:
      return "e.g., /address/zipCode";
  }
}

export function getPartitionKey(isQuickstart?: boolean): string {
  if (userContext.apiType !== "SQL" && userContext.apiType !== "Mongo") {
    return "";
  }
  if (userContext.features.partitionKeyDefault) {
    return userContext.apiType === "SQL" ? "/id" : "_id";
  }
  if (userContext.features.partitionKeyDefault2) {
    return userContext.apiType === "SQL" ? "/pk" : "pk";
  }
  if (isQuickstart) {
    return userContext.apiType === "SQL" ? "/categoryId" : "categoryId";
  }
  return "";
}

export function isFreeTierAccount(): boolean {
  return userContext.databaseAccount?.properties?.enableFreeTier;
}

export function UniqueKeysHeader(): JSX.Element {
  const tooltipContent =
    "Unique keys provide developers with the ability to add a layer of data integrity to their database. By creating a unique key policy when a container is created, you ensure the uniqueness of one or more values per partition key.";

  return (
    <Stack horizontal style={{ marginBottom: -2 }}>
      <Text className="panelTextBold" variant="small">
        Unique keys
      </Text>
      <TooltipHost directionalHint={DirectionalHint.bottomLeftEdge} content={tooltipContent}>
        <Icon iconName="Info" className="panelInfoIcon" tabIndex={0} ariaLabel={tooltipContent} />
      </TooltipHost>
    </Stack>
  );
}

export function shouldShowAnalyticalStoreOptions(): boolean {
  if (isFabricNative() || configContext.platform === Platform.Emulator) {
    return false;
  }

  switch (userContext.apiType) {
    case "SQL":
    case "Mongo":
      return true;
    default:
      return false;
  }
}

export function AnalyticalStoreHeader(): JSX.Element {
  const tooltipContent =
    "Enable analytical store capability to perform near real-time analytics on your operational data, without impacting the performance of transactional workloads.";
  return (
    <Stack horizontal style={{ marginBottom: -2 }}>
      <Text className="panelTextBold" variant="small">
        Analytical Store
      </Text>
      <TooltipHost directionalHint={DirectionalHint.bottomLeftEdge} content={tooltipContent}>
        <Icon iconName="Info" className="panelInfoIcon" tabIndex={0} ariaLabel={tooltipContent} />
      </TooltipHost>
    </Stack>
  );
}

export function AnalyticalStorageContent(): JSX.Element {
  return (
    <Text variant="small">
      Enable analytical store capability to perform near real-time analytics on your operational data, without impacting
      the performance of transactional workloads.{" "}
      <Link
        aria-label={Constants.ariaLabelForLearnMoreLink.AnalyticalStore}
        target="_blank"
        href="https://aka.ms/analytical-store-overview"
      >
        Learn more
      </Link>
    </Text>
  );
}

export function isSynapseLinkEnabled(): boolean {
  if (!userContext.databaseAccount) {
    return false;
  }

  const { properties } = userContext.databaseAccount;
  if (!properties) {
    return false;
  }

  if (properties.enableAnalyticalStorage) {
    return true;
  }

  return properties.capabilities?.some(
    (capability) => capability.name === Constants.CapabilityNames.EnableStorageAnalytics,
  );
}

export function scrollToSection(id: string): void {
  document.getElementById(id)?.scrollIntoView();
}

export function ContainerVectorPolicyTooltipContent(): JSX.Element {
  return (
    <Text variant="small">
      Describe any properties in your data that contain vectors, so that they can be made available for similarity
      queries.{" "}
      <Link target="_blank" href="https://aka.ms/CosmosDBVectorSetup">
        Learn more
      </Link>
    </Text>
  );
}

export function parseUniqueKeys(uniqueKeys: string[]): DataModels.UniqueKeyPolicy {
  if (uniqueKeys?.length === 0) {
    return undefined;
  }

  const uniqueKeyPolicy: DataModels.UniqueKeyPolicy = { uniqueKeys: [] };
  uniqueKeys.forEach((uniqueKey: string) => {
    if (uniqueKey) {
      const validPaths: string[] = uniqueKey.split(",")?.filter((path) => path?.length > 0);
      const trimmedPaths: string[] = validPaths?.map((path) => path.trim());
      if (trimmedPaths?.length > 0) {
        if (userContext.apiType === "Mongo") {
          trimmedPaths.map((path) => {
            const transformedPath = path.split(".").join("/");
            if (transformedPath[0] !== "/") {
              return "/" + transformedPath;
            }
            return transformedPath;
          });
        }
        uniqueKeyPolicy.uniqueKeys.push({ paths: trimmedPaths });
      }
    }
  });

  return uniqueKeyPolicy;
}

export const SharedDatabaseDefault: DataModels.IndexingPolicy = {
  indexingMode: "consistent",
  automatic: true,
  includedPaths: [],
  excludedPaths: [
    {
      path: "/*",
    },
  ],
};

export const FullTextPolicyDefault: DataModels.FullTextPolicy = {
  defaultLanguage: getFullTextLanguageOptions()[0].key as never,
  fullTextPaths: [],
};

export const AllPropertiesIndexed: DataModels.IndexingPolicy = {
  indexingMode: "consistent",
  automatic: true,
  includedPaths: [
    {
      path: "/*",
      indexes: [
        {
          kind: "Range",
          dataType: "Number",
          precision: -1,
        },
        {
          kind: "Range",
          dataType: "String",
          precision: -1,
        },
      ],
    },
  ],
  excludedPaths: [],
};
