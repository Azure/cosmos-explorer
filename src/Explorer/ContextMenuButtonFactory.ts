import AddCollectionIcon from "images/AddCollection.svg";
import AddSqlQueryIcon from "images/AddSqlQuery_16x16.svg";
import AddStoredProcedureIcon from "images/AddStoredProcedure.svg";
import AddTriggerIcon from "images/AddTrigger.svg";
import AddUdfIcon from "images/AddUdf.svg";
import DeleteCollectionIcon from "images/DeleteCollection.svg";
import DeleteDatabaseIcon from "images/DeleteDatabase.svg";
import DeleteSprocIcon from "images/DeleteSproc.svg";
import DeleteTriggerIcon from "images/DeleteTrigger.svg";
import DeleteUDFIcon from "images/DeleteUDF.svg";
import HostedTerminalIcon from "images/Hosted-Terminal.svg";
import * as ViewModels from "../Contracts/ViewModels";
import { DefaultAccountExperienceType } from "../DefaultAccountExperienceType";
import { userContext } from "../UserContext";
import { TreeNodeMenuItem } from "./Controls/TreeComponent/TreeComponent";
import Explorer from "./Explorer";
import StoredProcedure from "./Tree/StoredProcedure";
import Trigger from "./Tree/Trigger";
import UserDefinedFunction from "./Tree/UserDefinedFunction";

export interface CollectionContextMenuButtonParams {
  databaseId: string;
  collectionId: string;
}

export interface DatabaseContextMenuButtonParams {
  databaseId: string;
}
/**
 * New resource tree (in ReactJS)
 */
export class ResourceTreeContextMenuButtonFactory {
  public static createDatabaseContextMenu(container: Explorer): TreeNodeMenuItem[] {
    const items: TreeNodeMenuItem[] = [
      {
        iconSrc: AddCollectionIcon,
        onClick: () => container.onNewCollectionClicked(),
        label: container.addCollectionText(),
      },
    ];

    if (userContext.defaultExperience !== DefaultAccountExperienceType.Table) {
      items.push({
        iconSrc: DeleteDatabaseIcon,
        onClick: () => container.openDeleteDatabaseConfirmationPane(),
        label: container.deleteDatabaseText(),
        styleClass: "deleteDatabaseMenuItem",
      });
    }
    return items;
  }

  public static createCollectionContextMenuButton(
    container: Explorer,
    selectedCollection: ViewModels.Collection
  ): TreeNodeMenuItem[] {
    const items: TreeNodeMenuItem[] = [];
    if (userContext.apiType === "SQL" || userContext.apiType === "Gremlin") {
      items.push({
        iconSrc: AddSqlQueryIcon,
        onClick: () => selectedCollection && selectedCollection.onNewQueryClick(selectedCollection, null),
        label: "New SQL Query",
      });
    }

    if (userContext.apiType === "Mongo") {
      items.push({
        iconSrc: AddSqlQueryIcon,
        onClick: () => selectedCollection && selectedCollection.onNewMongoQueryClick(selectedCollection, null),
        label: "New Query",
      });

      items.push({
        iconSrc: HostedTerminalIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewMongoShellClick();
        },
        label: "New Shell",
      });
    }

    if (userContext.apiType === "SQL" || userContext.apiType === "Gremlin") {
      items.push({
        iconSrc: AddStoredProcedureIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewStoredProcedureClick(selectedCollection, null);
        },
        label: "New Stored Procedure",
      });

      items.push({
        iconSrc: AddUdfIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewUserDefinedFunctionClick(selectedCollection, null);
        },
        label: "New UDF",
      });

      items.push({
        iconSrc: AddTriggerIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewTriggerClick(selectedCollection, null);
        },
        label: "New Trigger",
      });
    }

    items.push({
      iconSrc: DeleteCollectionIcon,
      onClick: () => container.openDeleteCollectionConfirmationPane(),
      label: container.deleteCollectionText(),
      styleClass: "deleteCollectionMenuItem",
    });

    return items;
  }

  public static createStoreProcedureContextMenuItems(
    container: Explorer,
    storedProcedure: StoredProcedure
  ): TreeNodeMenuItem[] {
    if (userContext.apiType === "Cassandra") {
      return [];
    }

    return [
      {
        iconSrc: DeleteSprocIcon,
        onClick: () => storedProcedure.delete(),
        label: "Delete Store Procedure",
      },
    ];
  }

  public static createTriggerContextMenuItems(container: Explorer, trigger: Trigger): TreeNodeMenuItem[] {
    if (userContext.apiType === "Cassandra") {
      return [];
    }

    return [
      {
        iconSrc: DeleteTriggerIcon,
        onClick: () => trigger.delete(),
        label: "Delete Trigger",
      },
    ];
  }

  public static createUserDefinedFunctionContextMenuItems(
    container: Explorer,
    userDefinedFunction: UserDefinedFunction
  ): TreeNodeMenuItem[] {
    if (userContext.apiType === "Cassandra") {
      return [];
    }

    return [
      {
        iconSrc: DeleteUDFIcon,
        onClick: () => userDefinedFunction.delete(),
        label: "Delete User Defined Function",
      },
    ];
  }
}
