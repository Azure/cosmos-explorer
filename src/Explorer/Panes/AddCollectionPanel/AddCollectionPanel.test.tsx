import { shallow } from "enzyme";
import React from "react";
import Explorer from "../../Explorer";
import { AddCollectionPanel } from "./AddCollectionPanel";

const props = {
  explorer: new Explorer(),
};

describe("AddCollectionPanel", () => {
  it("should render Default properly", () => {
    const wrapper = shallow(<AddCollectionPanel {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
