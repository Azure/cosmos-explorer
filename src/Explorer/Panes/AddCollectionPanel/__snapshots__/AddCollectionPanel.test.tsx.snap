// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCollectionPanel should render Default properly 1`] = `
<form
  className="panelFormWrapper"
  id="panelContainer"
  onSubmit={[Function]}
>
  <div
    className="panelMainContent"
  >
    <Stack
      hidden={false}
      style={
        {
          "marginBottom": -2,
        }
      }
    >
      <Stack
        horizontal={true}
      >
        <span
          className="mandatoryStar"
        >
          * 
        </span>
        <Text
          className="panelTextBold"
          variant="small"
        >
          Database 
          id
        </Text>
        <StyledTooltipHostBase
          content="A database is analogous to a namespace. It is the unit of management for a set of containers."
          directionalHint={4}
        >
          <Icon
            ariaLabel="A database is analogous to a namespace. It is the unit of management for a set of containers."
            className="panelInfoIcon"
            iconName="Info"
            tabIndex={0}
          />
        </StyledTooltipHostBase>
      </Stack>
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <div
          role="radiogroup"
        >
          <input
            aria-checked={true}
            aria-label="Create new database"
            checked={true}
            className="panelRadioBtn"
            id="databaseCreateNew"
            name="databaseType"
            onChange={[Function]}
            role="radio"
            tabIndex={0}
            type="radio"
          />
          <span
            className="panelRadioBtnLabel"
          >
            Create new
          </span>
          <input
            aria-checked={false}
            aria-label="Use existing database"
            checked={false}
            className="panelRadioBtn"
            name="databaseType"
            onChange={[Function]}
            role="radio"
            tabIndex={0}
            type="radio"
          />
          <span
            className="panelRadioBtnLabel"
          >
            Use existing
          </span>
        </div>
      </Stack>
      <Stack
        className="panelGroupSpacing"
      >
        <input
          aria-label="New database id, Type a new database id"
          aria-required={true}
          autoComplete="off"
          className="panelTextField"
          id="newDatabaseId"
          name="newDatabaseId"
          onChange={[Function]}
          pattern="[^\\/?#\\\\]*[^\\/?# \\\\]"
          placeholder="Type a new database id"
          required={true}
          size={40}
          tabIndex={0}
          title="May not end with space nor contain characters '\\' '/' '#' '?'"
          type="text"
          value=""
        />
        <Stack
          horizontal={true}
        >
          <StyledCheckboxBase
            checked={false}
            label="Share throughput across containers"
            onChange={[Function]}
            styles={
              {
                "checkbox": {
                  "height": 12,
                  "width": 12,
                },
                "label": {
                  "alignItems": "center",
                  "padding": 0,
                },
                "text": {
                  "fontSize": 12,
                },
              }
            }
          />
          <StyledTooltipHostBase
            content="Throughput configured at the database level will be shared across all containers within the database."
            directionalHint={4}
          >
            <Icon
              ariaLabel="Throughput configured at the database level will be shared across all containers within the database."
              className="panelInfoIcon"
              iconName="Info"
              tabIndex={0}
            />
          </StyledTooltipHostBase>
        </Stack>
      </Stack>
      <Separator
        className="panelSeparator"
        style={
          {
            "marginBottom": -4,
            "marginTop": -4,
          }
        }
      />
    </Stack>
    <Stack>
      <Stack
        horizontal={true}
        style={
          {
            "marginBottom": 1,
            "marginTop": -5,
          }
        }
      >
        <span
          className="mandatoryStar"
        >
          * 
        </span>
        <Text
          className="panelTextBold"
          variant="small"
        >
          Container id
        </Text>
        <StyledTooltipHostBase
          content="Unique identifier for the container and used for id-based routing through REST and all SDKs."
          directionalHint={4}
        >
          <Icon
            ariaLabel="Unique identifier for the container and used for id-based routing through REST and all SDKs."
            className="panelInfoIcon"
            iconName="Info"
            role="button"
            tabIndex={0}
          />
        </StyledTooltipHostBase>
      </Stack>
      <input
        aria-label="Container id, Example Container1"
        aria-required={true}
        autoComplete="off"
        className="panelTextField"
        id="collectionId"
        name="collectionId"
        onChange={[Function]}
        pattern="[^\\/?#\\\\]*[^\\/?# \\\\]"
        placeholder="e.g., Container1"
        required={true}
        size={40}
        title="May not end with space nor contain characters '\\' '/' '#' '?'"
        type="text"
        value=""
      />
      <Separator
        className="panelSeparator"
        style={
          {
            "marginBottom": -5,
            "marginTop": -5,
          }
        }
      />
    </Stack>
    <Stack>
      <Stack
        horizontal={true}
        style={
          {
            "marginBottom": -4,
            "marginTop": -5,
          }
        }
      >
        <span
          className="mandatoryStar"
        >
          * 
        </span>
        <Text
          className="panelTextBold"
          variant="small"
        >
          Partition key
        </Text>
        <StyledTooltipHostBase
          content="The partition key is used to automatically distribute data across partitions for scalability. Choose a property in your JSON document that has a wide range of values and evenly distributes request volume. For small read-heavy workloads or write-heavy workloads of any size, id is often a good choice."
          directionalHint={4}
        >
          <Icon
            ariaLabel="The partition key is used to automatically distribute data across partitions for scalability. Choose a property in your JSON document that has a wide range of values and evenly distributes request volume. For small read-heavy workloads or write-heavy workloads of any size, id is often a good choice."
            className="panelInfoIcon"
            iconName="Info"
            tabIndex={0}
          />
        </StyledTooltipHostBase>
      </Stack>
      <Text
        variant="small"
      />
      <input
        aria-label="Partition key"
        aria-required={true}
        className="panelTextField"
        id="addCollection-partitionKeyValue"
        onChange={[Function]}
        pattern=".*"
        placeholder="Required - first partition key e.g., /TenantId"
        required={true}
        size={40}
        title=""
        type="text"
        value=""
      />
      <Stack
        className="panelGroupSpacing"
      >
        <CustomizedDefaultButton
          disabled={false}
          hidden={false}
          onClick={[Function]}
          styles={
            {
              "label": {
                "fontSize": 12,
              },
              "root": {
                "height": 30,
                "padding": 0,
                "width": 200,
              },
            }
          }
        >
          Add hierarchical partition key
        </CustomizedDefaultButton>
      </Stack>
      <Separator
        className="panelSeparator"
        style={
          {
            "marginBottom": -4,
            "marginTop": 2,
          }
        }
      />
    </Stack>
    <ThroughputInput
      isDatabase={false}
      isSharded={true}
      onCostAcknowledgeChange={[Function]}
      setIsAutoscale={[Function]}
      setIsThroughputCapExceeded={[Function]}
      setThroughputValue={[Function]}
    />
    <Stack
      style={
        {
          "marginBottom": -4,
          "marginTop": -2,
        }
      }
    >
      <Stack
        horizontal={true}
        style={
          {
            "marginBottom": -2,
          }
        }
      >
        <Text
          className="panelTextBold"
          variant="small"
        >
          Unique keys
        </Text>
        <StyledTooltipHostBase
          content="Unique keys provide developers with the ability to add a layer of data integrity to their database. By creating a unique key policy when a container is created, you ensure the uniqueness of one or more values per partition key."
          directionalHint={4}
        >
          <Icon
            ariaLabel="Unique keys provide developers with the ability to add a layer of data integrity to their database. By creating a unique key policy when a container is created, you ensure the uniqueness of one or more values per partition key."
            className="panelInfoIcon"
            iconName="Info"
            tabIndex={0}
          />
        </StyledTooltipHostBase>
      </Stack>
      <CustomizedActionButton
        iconProps={
          {
            "iconName": "Add",
          }
        }
        onClick={[Function]}
        styles={
          {
            "label": {
              "fontSize": 12,
            },
            "root": {
              "padding": 0,
            },
          }
        }
      >
        Add unique key
      </CustomizedActionButton>
    </Stack>
    <Separator
      className="panelSeparator"
      style={
        {
          "marginBottom": -4,
          "marginTop": -15,
        }
      }
    />
    <Stack
      className="panelGroupSpacing"
      style={
        {
          "marginTop": -4,
        }
      }
    >
      <Text
        className="panelTextBold"
        variant="small"
      >
        <Stack
          horizontal={true}
          style={
            {
              "marginBottom": -2,
            }
          }
        >
          <Text
            className="panelTextBold"
            variant="small"
          >
            Analytical Store
          </Text>
          <StyledTooltipHostBase
            content="Enable analytical store capability to perform near real-time analytics on your operational data, without impacting the performance of transactional workloads."
            directionalHint={4}
          >
            <Icon
              ariaLabel="Enable analytical store capability to perform near real-time analytics on your operational data, without impacting the performance of transactional workloads."
              className="panelInfoIcon"
              iconName="Info"
              tabIndex={0}
            />
          </StyledTooltipHostBase>
        </Stack>
      </Text>
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <div
          role="radiogroup"
        >
          <input
            aria-checked={false}
            aria-label="Enable analytical store"
            checked={false}
            className="panelRadioBtn"
            disabled={true}
            id="enableAnalyticalStoreBtn"
            name="analyticalStore"
            onChange={[Function]}
            role="radio"
            tabIndex={0}
            type="radio"
          />
          <span
            className="panelRadioBtnLabel"
          >
            On
          </span>
          <input
            aria-checked={true}
            aria-label="Disable analytical store"
            checked={true}
            className="panelRadioBtn"
            disabled={true}
            id="disableAnalyticalStoreBtn"
            name="analyticalStore"
            onChange={[Function]}
            role="radio"
            tabIndex={0}
            type="radio"
          />
          <span
            className="panelRadioBtnLabel"
          >
            Off
          </span>
        </div>
      </Stack>
      <Stack
        className="panelGroupSpacing"
      >
        <Text
          variant="small"
        >
          Azure Synapse Link is required for creating an analytical store
           
          container
          . Enable Synapse Link for this Cosmos DB account. 
          <br />
          <StyledLinkBase
            aria-label="Learn more about Azure Synapse Link."
            className="capacitycalculator-link"
            href="https://aka.ms/cosmosdb-synapselink"
            target="_blank"
          >
            Learn more
          </StyledLinkBase>
        </Text>
        <CustomizedDefaultButton
          onClick={[Function]}
          style={
            {
              "height": 27,
              "width": 80,
            }
          }
          styles={
            {
              "label": {
                "fontSize": 12,
              },
            }
          }
          text="Enable"
        />
      </Stack>
    </Stack>
    <Stack>
      <CollapsibleSectionComponent
        isExpandedByDefault={false}
        onExpand={[Function]}
        title="Container Full Text Search Policy"
      >
        <Stack
          id="collapsibleFullTextPolicySectionContent"
          styles={
            {
              "root": {
                "position": "relative",
              },
            }
          }
        >
          <Stack
            styles={
              {
                "root": {
                  "paddingLeft": 40,
                },
              }
            }
          >
            <FullTextPoliciesComponent
              fullTextPolicy={
                {
                  "defaultLanguage": "en-US",
                  "fullTextPaths": [],
                }
              }
              onFullTextPathChange={[Function]}
            />
          </Stack>
        </Stack>
      </CollapsibleSectionComponent>
    </Stack>
    <CollapsibleSectionComponent
      isExpandedByDefault={false}
      onExpand={[Function]}
      title="Advanced"
    >
      <Stack
        className="panelGroupSpacing"
        id="collapsibleAdvancedSectionContent"
      >
        <Stack
          className="panelGroupSpacing"
        >
          <StyledCheckboxBase
            checked={false}
            label="My application uses an older Cosmos .NET or Java SDK version (.NET V1 or Java V2)"
            onChange={[Function]}
            styles={
              {
                "checkbox": {
                  "height": 12,
                  "width": 12,
                },
                "label": {
                  "alignItems": "center",
                  "padding": 0,
                  "whiteSpace": "break-spaces",
                  "wordWrap": "break-word",
                },
                "text": {
                  "fontSize": 12,
                },
              }
            }
          />
          <Text
            variant="small"
          >
            <Icon
              className="removeIcon"
              iconName="InfoSolid"
            />
             To ensure compatibility with older SDKs, the created container will use a legacy partitioning scheme that supports partition key values of size only up to 101 bytes. If this is enabled, you will not be able to use hierarchical partition keys.
             
            <StyledLinkBase
              href="https://aka.ms/cosmos-large-pk"
              target="_blank"
            >
              Learn more
            </StyledLinkBase>
          </Text>
        </Stack>
      </Stack>
    </CollapsibleSectionComponent>
  </div>
  <PanelFooterComponent
    buttonLabel="OK"
    isButtonDisabled={false}
  />
</form>
`;
